### PR TITLE
Update non-platform dependencies to most modern possible

### PR DESCRIPTION
**Pull request by DylanLacey**
To take advantage of any security or performance bug-fixes, as well as increase compatibility with current libraries, update everything besides core `next` and `react` packages to the latest possible version.

*UPDATED WITH CHANGES*
*caniuse-lite*
- Updated browser config

*compareVersions*
- Updated to 6.1.0
- Updated import

*evergreen-ui*
- Updated to 7.1.9
- psuedoselectors in themes now need to be placed in a `selector` object, so did that

*eslint*
- Moved config from `package.json` to `eslintrc.yaml` to allow rules to be commented
- Added `eslint-config-next`
- Disabled several rules causing existing code to cause linting errors
- Added TODO to correct the above at a later point.

*jest*
- Updated to 28.7.0
- Migrated config from `jest.config.js` to `jest.config.ts`
- Installed the latest `jsdom` (See https://github.com/jestjs/jest/issues/6766)

*next-mdx-remote*
- Updated to 4.0.1
- Moved `next-sitemap.js` to `next-sitemap.config.js` as required by update

*next-sitemap*
- Updated to 4.2.3
- `components` definition in `MDXRenderer.tsx` were causing the build type issues; Resolve by wrapping all entries in anonymous components. I'm not great at react; There's probably a better way

*react-scripts*
- Updated to 5.0.1
- Yarn is _very_ mad about the wealth of outdated dependencies. Suggest we replace `react-scripts` with "raw" Next... Or migrate to Vite! 🧌

*@types/whatever*
- Moved any in `dependencies` to `devDependencies`
- Updated where possible
- Updated `@types/react` and `@types/react-dom` and set a `resolution` in `package.json`; Something was pulling in version `*` of both, making the build mad about `Components` not being `JSXComponents`. This will probably go away once we migrate to React 18.

*yaml*
- Updated to 2.4.3
- Removed `@types/yaml` as it's now packaged in

*NOT UPDATED*
* `react-gfm` broke silently when updated to 4.0.0

*REMOVED DUE TO APPARENTLY BEING UNUSED*
* `react-markdown`
* `lowlight`
* `rehype-highlight`

### CODE REVIEW

**Review by AucaCoyan — APPROVED** [review]
All good!

**Comment by AucaCoyan** [comment]
Just a tiny note: I'm a friend of using more rust tools, so maybe in the future we can make a PR to start using [biome](https://biomejs.dev/)
But not today, don't hurry!

**Comment by DylanLacey** [comment]
I _think_ this update to Yarn Modern should un-fsck the dependency management; It did when I tested within my own repo.

**Comment by AucaCoyan** [comment]
Mmm I've never used yarn v4 on vercel. Does anybody has experience with that?
It would be sad that it could be deployed locally with the latest, but the production version is used with yarn v1 😢

**Comment by DylanLacey** [comment]
LOL Vercel only supports Yarn 2+ for Static builds, not Serverless Functions or SSR. It's only 4 years old, after all... 👀 : https://vercel.com/guides/does-vercel-support-yarn-2

AFAICT the hub doesn't use SSR, but I'm not sure about Serverless functions?

**Comment by arabello** [comment]
Jumping in to say that I successfully tried the latest changes locally and it seems to work. I am not sure if updating all the dependencies is necessary for security or performance reasons as this project is quite small, but as long as the deploy works and no functionalities are affected I see no reasons why we shouldn't.

TL;DR After merging it, let's keep an eye on deployment and smoke test it.

> I've never used yarn v4 on vercel
> AFAICT the hub doesn't use SSR, but I'm not sure about Serverless functions?

I am not sure to understand why Vercel is relevant to us: the hub is [built and deployed](https://github.com/espanso/hub-frontend/blob/main/.github/workflows/deploy.yml) as a static website on GH Pages. Am I missing something? (it's been a while for me away from the hub, I might need to catch up things)

**Comment by AucaCoyan** [comment]
Oh, it was my wrong assumption. Is there any reason why we don't use vercel? it has nice integration with prs, you can view the page deployed even before merging the pr (like the one we have on `website`).

**Comment by arabello** [comment]
> Is there any reason why we don't use vercel?

No specific reason at all: we started the hub as simple as possible to minimize effort against provided value and being the solo dev I didn't need additional stuff.

We might consider it as the hub evolves.

**Comment by AucaCoyan** [comment]
Great!
We can write down in the readme or a contributing md how to develop the site, so we have guidelines for new people :smile:

**Comment by DylanLacey** [comment]
Yeah, I've made some notes while updating the deps; I'll turn them into a Contributing.md in the next few days.  We can merge as soon as someone does a review.

(FWIW I updated the deps basically because they were outdated and it seemed like a good way to introduce myself to the codebase)

**Review by AucaCoyan — APPROVED** [review]

**Comment by AucaCoyan** [comment]
Hey Dylan, where are we in this PR?

**Comment by AucaCoyan** [comment]
Sup @DylanLacey , do you have any pending task here? 😄

**Comment by DylanLacey** [comment]
Hey @AucaCoyan ! Sorry for the delay; No, this is ready to merge, I just don't have permissions.
